src/hospitalgame/NPC/Doctor.java

```diff
@@ -8,24 +12,42 @@
  * @author Lars Bjerregaard Jørgensen
  * @author Robert Francisti
  */
-public class Doctor extends NPC{
-    
+public class Doctor extends NPC {
+
     /**
      * Calls the NPC constructor through the super
-     * @param currentRoom the room that the NPC is in right now
-     * @param name name of the NPC  
-     * @param description decription of the NPC 
+     *
+     * @param name name of the NPC
+     * @param description decription of the NPC
      */
-    public Doctor (Room currentRoom, String name, String description){
-        super(currentRoom, name, description);
+    public Doctor(String name, String description) {
+        super(name, description);
     }
 
     /**
      * Overrides the abstract method in NPC
+     *
      * @param player is the player object
      */
     @Override
     public void interact(Player player) {
-        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
-    } 
+        BloodType bloodType = player.getBloodType();
+
+        BloodBag[] bloodBags = (BloodBag[]) player.getItemsByName(ItemName.BLOODBAG);
+        for (BloodBag bloodBag : bloodBags) {
+            if (!bloodType.canTransfuse(bloodBag.getBloodType())) {
+                System.out.println("You lost the game.");
+                return;
+
+            }
+
+        }
+        int points = (int) player.getBloodAmount();
+        for (BloodBag bloodBag : bloodBags) {
```

Comment:
int points = (int) player.getBloodAmount();

         for (BloodBag bloodBag : bloodBags) {
              if (!bloodType.canTransfuse(bloodBag.getBloodType())) {
                  System.out.println("You lost the game.");
                  return;
              }else{
                   points += bloodBag.getBonusPoints();
                }

         }